lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -822,7 +822,7 @@ public function count($criteria = array())
             ? $this->expandCriteriaParameters($criteria)
             : $this->expandParameters($criteria);
 
-        return $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
+        return (int) $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
```

Comment:
In that case I think we should check that situation, keeping the current behavior on fail:
```php
// Check if `fetchColumn()` response has failed (`false`).
if (false === $count = $this->conn->executeQuery($sql, $params, $types)->fetchColumn()) {
    return $count;
}

return (int) $count;